### DISCUSSION TITLE
V4 Hardware

### DISCUSSION THREAD

**Discussion by CaitSith2** [sanni/cartreader#284] [General Chat]
I noticed that the wiki updated to show all of the Hardware revisions, and see that V4 hardware seems to be moving towards being a board that doesn't function as a dumper directly, but rather, you make adapter boards for the various system cartridges you wish to be able to dump.

**Comment by sanni** [comment]
Yeah, the idea of the V4 hardware is not to make it better but to make it more simple. So no SMD components and all the modules are plugged into the main PCB with pin headers so they can be removed at any time for easier trouble shooting. And there will only be one cart slot attached at a time too. Plus there won't be any configuration switches, everything will be setup by different wirings on the cart slot adapter PCBs. Just a 3V/5V switch on some adapters for compatibility reasons.

There is also this neat little trick where you don't need the snesCIC if you bridge pin 55 and 24 on the SNES cart slot. So that will be used here too.

I have ordered these PCBs for a first test. If they work I'll upload the KiCad and Gerber files for everyone to test.

Main PCB, backside does not have any components:
![449494946464](https://user-images.githubusercontent.com/19169494/140276843-ee743ed4-bb37-43a8-95b5-ab66997fa8f7.jpg)

Those are the modules from Aliexpress that will be used:
![IMG_0614](https://user-images.githubusercontent.com/19169494/140276840-f9779172-6db4-4273-9d94-3a42374e664a.jpg)

I did already add support for the LCD with the rotary encoder and it works great and makes selecting stuff much faster:
![3d86001257d21fc93c606308bebd5be911aecbab](https://user-images.githubusercontent.com/19169494/140277490-96a25398-f068-4e3d-be1e-41c937a42eb8.jpeg)

I really hope these changes allow it to be built more easily by beginners or people with not so much time. Especially since you can build it in steps, like first just the main PCB to see if you get output on the LCD, then you can build one cart slot adapter see if that works then another one and so on. To the contrary with the V3 hardware usually as a beginner you need several hours, lose concentration and end up with something that doesn't work with no clue why.

**Comment by CaitSith2** [comment]
That SMD looking pad on the snes part of the adapter PCB,  I am guessing is a simple solder jumper in case you are building that adapter board strictly for snes carts rather than for use with previously built adapters.

Also, what is up with the idea behind bridging pin 24 and 55, and how does that bypass the need for a cic?  That works even for SA1?

**Comment by sanni** [comment]
Yes to your first conclusion. And about the snesCIC: If you bridge pin 24 and 55 on the cart slot or the snesCIC pads the CIC inside the cartridge just talks to itself and unlocks itself. 
In my tests I found that it unlocks SA1 even better than with a snesCIC. This method also works in a real console by removing the CIC there and shorting the respective pins.

![d4188402e97ac828ccc832e851608deae4a63961](https://user-images.githubusercontent.com/19169494/140328966-a473a53e-d13d-4d94-8b22-966205b8fdae.jpeg)

**Reply by CaitSith2** [reply]
I guess in this case, the CIC clock won't be as sensitive since the cartridge is strictly talking to itself.

I guess on the V3 boards, I could bridge 6 and 7 of the PIC chip mounting spot to effectively bridge 24 and 55.

**Reply by fakkuyuu** [reply]
I'm trying to make a V3. If I use this bridging method, do I need to add a 100nf capacitor next to it?

**Comment by CaitSith2** [comment]
That step in designing your own adapters where you let free router do the auto-routing,  I actually enjoy manually routing boards.  I see each board to be routed as a unique puzzle.

**Comment by giklette** [comment]
Why need we desolder the fuse for the V4 ?
If I do it, the Arduino is no more discover in Windows, so I need to solder fuse, then I can find the Ardiuno and flash it.
So can I replace the fuse or may be I should test all my solders ?

**Comment by sanni** [comment]
Background:    
The Arduino normally runs at 5V. But some game cartridges need 3.3V. If you just supply 3.3V to the VCC pins of the game cartridge the address lines will still be at a 5V level. So to get everything to a 3.3V level you either need voltage level shifters for each and every cartridge pin or you can just run the whole Arduino at 3.3V. I choose the second option for HW4 since HW4 is meant to be as simple as possible with the least amount of parts needed to function.    

Fuse:    
The fuse gets removed to disconnect the 5V USB port from the Arduino. The 5V from the USB port then gets routed through the battery module, through the LCD module and through the cartridge adapter module on the Cart Reader PCB until it finally comes back to the Arduino either as 5V or 3.3V depending on the inserted adapter module. That way the Arduino can run at either 5V or 3.3V.    

Flashing:    
It is normal that you can't flash the stand-alone Arduino if the fuse is removed.     
You also can't flash the completed Cart Reader if the battery module is missing or if the battery is not inserted or if the Cart Reader is not switched on or if there is no cartridge adapter module installed.     
You can however replace the battery module or the cart adapter module with a wire if you don't have them handy at this moment, see the Wiki troubleshooting article for more details.

**Reply by giklette** [reply]
it's super clear, thank you a lot
